src/testutils.rs

```diff
@@ -33,16 +34,21 @@ impl Default for TestEnvironment {
             _temp_dir: tmp_dir,
             env_root,
             home_dir,
+            command_number: 0,
         }
     }
 }
 
 impl TestEnvironment {
-    pub fn jj_cmd(&self, current_dir: &Path, args: &[&str]) -> assert_cmd::Command {
+    pub fn jj_cmd(&mut self, current_dir: &Path, args: &[&str]) -> assert_cmd::Command {
         let mut cmd = assert_cmd::Command::cargo_bin("jj").unwrap();
         cmd.current_dir(current_dir);
         cmd.args(args);
         cmd.env("HOME", self.home_dir.to_str().unwrap());
+        let timestamp = chrono::DateTime::parse_from_rfc3339("2001-02-03T04:05:06+07:00").unwrap();
+        self.command_number += 1;
```

Comment:
An alternative would be to wrap `command_number` in a `RefCell`, which seems reasonable since it's unlikely you would be mutating it simultaneously in practice. Then you don't need to clone `current_dir`.

Reply:
Good point, I'll make that change.